internal/maintenance/job_scheduler_test.go

```diff
@@ -139,6 +148,50 @@ func TestJobScheduler(t *testing.T) {
 		requireJobStateUnchanged(t, bundle.exec, retryableJob3) // still retryable
 	})
 
+	t.Run("MovesUniqueKeyConflictingJobsToDiscarded", func(t *testing.T) {
```

Comment:
I realized I don't have corresponding test changes for the underlying query in the `riverdrivertest`, lmk if you think these details should be duplicated over there.